### PR TITLE
Add CI and license

### LINKED ISSUE #14: CI etc
Currently, this repo doesn't have CI and stuff.

The usual way in which I set up a repo is to first use a default repo provided by `PkgTemplates.jl`, and then play around with the CI etc to give me what I need. This could be done for this repo.

If not that, it should be possible to copy + paste the github actions from AbstractGPs into this repo, and modify details.

Also, a license (we usually use MIT) would be a good idea.

### PR DESCRIPTION
**Pull request by rossviljoen**
closes #14

### CODE REVIEW

**Comment by rossviljoen** [comment]
Oh, CI won't run unless GPLikelihoods is registered :smile: 

waiting for https://github.com/JuliaGaussianProcesses/GPLikelihoods.jl/issues/33

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  test:
+    name: Julia ${{ matrix.version }} - ${{ matrix.os }} - ${{ matrix.arch }} - ${{ github.event_name }}
+    runs-on: ${{ matrix.os }}
+    continue-on-error: ${{ matrix.version == 'nightly' }}
+    strategy:
+      fail-fast: false
+      matrix:
+        version:
+          - '1'
+          - '1.3'
```
Comment (willtebbutt) [inline .github/workflows/CI.yml:19]:
```suggestion
```
I wouldn't bother commiting yourself to 1.3. To be honest, I'd require at least 1.6

File: .github/workflows/CI.yml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+          - 'nightly'
+        os:
+          - ubuntu-latest
+          - macOS-latest
+          - windows-latest
```
Comment (willtebbutt) [inline .github/workflows/CI.yml:24]:
```suggestion
```
My inclination is not to bother testing on all OSs unless you're doing something that you think is OS-specific. The additional time spent waiting on CI is generally not worth it in my experience.

**Review by willtebbutt — APPROVED** [review]
LGTM. Just a couple of suggestions to keep a reasonable in terms of how much CI you're running.

**Comment by rossviljoen** [comment]
Seem like the test failure is unrelated?

**Comment by willtebbutt** [comment]
> Seems like the test failure is unrelated?

Yeah. As a rule, nightly fails on a regular bases, so I wouldn't worry about it. We merge when nightly fails all of the time.

**Pull request merged**